TLM/TLM/Util/RoadSelectionUtil.cs

```diff
@@ -0,0 +1,173 @@
+
+namespace TrafficManager.Util {
+    using System;
+    using System.Collections.Generic;
+    using System.Reflection;
+    using ColossalFramework;
+    using ICities;
+    using CSUtil.Commons;
+    using static TrafficManager.Util.Shortcuts;
+    using static NetAdjust;
+    using static SegmentTraverser;
+
+    public class RoadSelectionUtil {
+        /// instance of singleton
+        public static RoadSelectionUtil Instance { get; private set; }
+
+        public RoadSelectionUtil() : base() {
+            Instance = this;
+        }
+
+        public static void Release() {
+            Instance.OnChanged = null;
+            Instance = null;
+        }
+
+        /// <summary>
+        /// Copies road name from <paramref name="sourceSegmentID"/> to <paramref name="targetSegmentID"/>
+        /// This will join target segment to the same road as source segment.
+        /// </summary>
+        public static void CopySegmentName(ushort sourceSegmentID, ushort targetSegmentID) {
+            SimulationManager.instance.AddAction(delegate () {
+                string sourceName = NetManager.instance.GetSegmentName(sourceSegmentID);
+                string targetName = NetManager.instance.GetSegmentName(targetSegmentID);
+                if (sourceName != targetName) {
+                    NetManager.instance.SetSegmentName(targetSegmentID, sourceName);
+                }
+            });
+        }
+
+        /// <summary>
+        /// Joins all segments from <paramref name="segmentIDs"/> to the road
+        /// containing <paramref name="segmentId"/>
+        /// </summary>
+        public static void SetRoad(ushort segmentId, IEnumerable<ushort> segmentIDs) {
+            foreach(var targetSegmentID in segmentIDs) {
+                if(targetSegmentID!= segmentId&& targetSegmentID != 0) {
+                    CopySegmentName(segmentId, targetSegmentID);
+                }
+            }
+        }
+
+        internal const int NETADJUST_INFO_INDEX = 2;
+        internal static bool IsNetAdjustMode() =>
+            IsNetAdjustMode(InfoManager.instance.CurrentMode, (int)InfoManager.instance.CurrentSubMode);
+
+        internal static bool IsNetAdjustMode(InfoManager.InfoMode mode, int subModeIndex) =>
+            (mode == InfoManager.InfoMode.TrafficRoutes) &
+            (subModeIndex == NETADJUST_INFO_INDEX);
+
+        public int Length => GetPath()?.m_size ?? 0;
+
+        /// <summary>
+        /// Creates a list of selected segment IDs.
+        /// Modifying the returned list has no side effects.
+        /// </summary>
+        public List<ushort> Selection {
+            get {
+                if (Length > 0) {
+                    FastList<ushort> path = GetPath();
```

Comment:
Isn't it null possible here? If not why do you use `GetPath()?` in `Length` property?